src/base/object_representation.h

```diff
@@ -418,6 +382,42 @@ OR_PUT_DOUBLE (char *ptr, double val)
     OR_PUT_SHORT (((char *) (ptr)) + OR_LOG_LSA_OFFSET, -1); \
   } while (0)
 
+/* VARIABLE OFFSET ACCESSORS */
+
+#define OR_PUT_OFFSET_INTERNAL(ptr, val, offset_size) \
+  do { \
+    if (offset_size == OR_BYTE_SIZE) \
+      { \
+        OR_PUT_BYTE(ptr, val); \
+      } \
+    else if (offset_size == OR_SHORT_SIZE) \
+      { \
+        OR_PUT_SHORT(ptr, val); \
+      } \
+    else if (offset_size == OR_INT_SIZE) \
+      { \
+        OR_PUT_INT(ptr, val); \
+      } \
+  } while (0)
```

Comment:
Let's check the int type first. 
The same applies to OR_GET_OFFSET_INTERNAL().

Reply:
OR_PUT_OFFSET and OR_GET_OFFSET change to direct calls to OR_PUT_BIG_VAR_OFFSET and OR_GET_BIG_VAR_OFFSET respectively.
This way we don't have to change OR_INT_SIZE to compare first.